exchanges/orderbook/orderbook.go

```diff
@@ -271,9 +271,9 @@ var dsc = func(current Item, previous Item) error {
 }
 
 // checkAlignment validates full orderbook
-func checkAlignment(depth Items, fundingRate, priceDuplication, isIDAligned bool, c checker) error {
+func checkAlignment(depth Items, fundingRate, priceDuplication, isIDAligned bool, c checker, exch string) error {
 	for i := range depth {
-		if depth[i].Price == 0 {
+		if depth[i].Price == 0 && !fundingRate && exch != "Bitfinex" /* funding rate can be 0 it seems on Bitfinex */ {
```

Comment:
I'm concerned about this if check and comment. I think it will allow for things to slip by unnoticed with future changes in code. Yes `fundingRate` is only applied in Bitfinex's case right now, but if that ever changes, then funding rates of zero are allowed and people are not going to go out and look at this specific snippet of code if a bool ever changes from false to true in a wrapper function.

```go
func TestCheckAlignment(t *testing.T) {
	t.Parallel()
	itemWithFunding := Items{
		{
			Amount: 1337,
			Price:  0,
			Period: 1337,
		},
	}
	err := checkAlignment(itemWithFunding, true, true, false, dsc, "Bitfinex")
	if err != nil {
		t.Error(err)
	}
	err = checkAlignment(itemWithFunding, false, true, false, dsc, "Bitfinex")
	if err != nil {
		t.Error(err)
	}
	err = checkAlignment(itemWithFunding, true, true, false, dsc, "Binance")
	if err != nil {
		t.Error(err)
	}
}
```

`--- PASS: TestCheckAlignment`
I think it could be rewritten.
This isn't nearly as neat and could be improved upon, but I'm trying to instead only pass the specific instance for Bitfinex and pass nothing else:
```go
		switch {
		case exch == "Bitfinex" && depth[i].Price == 0 && fundingRate:
		case depth[i].Price == 0:
			return errPriceNotSet
		}
```

Reply:
🦅